CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+else... please read on.
+
+GitHub issues is the preferred mechanism for [bug reports](#bug-reports),
+[feature requests](#feature-requests), [change requests](#change-requests), and
+[submitting pull requests](#submitting-pull-requests).  However, please respect
+the following general restrictions:
+
+* Before submitting an issue, please **search** if one already exists.  Help us
+  keep duplicate issues to a minimum by checking if someone has already
+  reported your problem or idea.
+
+* Please keep discussions **on topic** and **respect** the opinions of others.
+
+
+### Bug reports
+
+A bug is a _demonstrable problem_ that is caused by code in this repository.
+We appreciate good bug reports!
+
+Guidelines for bug reports:
+
+1. **Search for the issue** &mdash; check if the issue has already been
+   reported.
+
+2. **Check if the issue has been fixed** &mdash; try to reproduce it using the
+   latest `develop` branch or look for [closed issues](http://github.com/bloomberg/blpapi-http/issues?q=is%3Aissue+is%3Aclosed).
+
+3. **Reduce the problem** &mdash; minimizing the dependencies that while still
+   being able to reproduce the problem is awesome.
+
+It is important to include the details of your environment and [what you have
+tried](http://whathaveyoutried.com).  Great bug reports shouldn't leave others
+needing to ask for further information; however, sometimes that does happen -
+not everyone is perfect.
+
+
+### Feature requests
+
+Features requests are always welcome!  Thanks for wanting to help out.  Before
+you submit:
+
+1. **Search around** if your feature is already being worked on or is already
+   listed in our [milestones](#milestones).
+
+2. Think about whether your idea fits with the scope of the project and its
+   direction.
+
+3. Make a strong case to convince the core contributors of the merits of the
+   feature.  Please provide as much detail as possible explaining the context
+   and use case.
+
+
+### Change requests
+
+Change requests cover the internal architectural and functional change to how
+blpapi-http works.  This includes new dependencies (or removal of them),
+refactorings, or an improvements to the system.
+
+1. **Search around** and make sure a similar request doesn't already exist.
+
+2. Please take the time to think about the best way to make a case for the
+   change.  Are you sure it is not a [bug report](#bug-reports) or a [feature
+   request](#feature-requests)?  What is the context and what precise problem
+   is it solving; or is it many?  Why this change that you are suggesting
+   making (or replacing) what is already there better?  Does it fit within the
+   direction of the project?
+
+
+## Submitting pull requests
+
+We love pull requests!  If your are raising a PR for something which doesn't
+have an open issue, please think carefully about [raising an
+issue](#reporting-an-issue) for which your PR can close - especially if it is a
```

Comment:
The phrasing "thing for which it can affect" sounds wrong to me -- specifically the "for which" seems off to me -- but I can't articulate why, so if you disagree I'm fine with no change here.